pkg/controller/machine/controller.go

```diff
@@ -120,11 +121,30 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+
+	// Set the ownerRef with foreground deletion if there is a linked cluster.
+	if cluster != nil && len(m.OwnerReferences) == 0 {
```

Comment:
This will currently orphan any machines that are part of a MachineSet or a MachineDeployment, since they will have an OwnerRef for the associated MachineSet/MachineDeployment. That said, I think this is probably the right approach and we should also add an OwnerRef (and do label matching) for MachineSets/MachineDeployments.

Reply:
I'll followup with another PR for label matching for MachineSet/Deployments, when we set the ownerRef there, deleting a cluster will also delete Set/Deployments which should be correct, unless I'm missing something 😄